torch/csrc/jit/tensorexpr/schedule.cpp

```diff
@@ -733,10 +731,15 @@ SplitAxisWithTail::SplitAxisWithTail(
   // TODO: support factor_on_inner == false;
   CHECK(factor_on_inner) << "only factor_on_inner = True is supported for now";
 
-  int size = this->stop() - this->start();
-  int split_count = size / factor;
-  int tail_size = size % factor;
-  int output_group_count = (tail_size > 0) ? 2 : 1;
+  auto size = this->stop() - this->start();
+  int output_group_count = 2;
+  auto split_count = size / factor;
+  auto tail_size = size % factor;
+  auto tail_imm = tail_size.AsNode<IntImm>();
+  bool has_tail = !tail_imm || (tail_imm->value() != 0);
```

Comment:
Do we have a test that show this reliably trigger for the no-tail case with static shapes? 

Because we need to evaluate the static shape one to know for sure the tail size is zero.

Reply:
Let me check, if not I'll add one

Reply:
Oh, actually you're right, since we don't eagerly constant fold `tail_imm` is never going to be a constant.

Reply:
Fixed this.  Thank you for catching it!